routers/web/admin/users.go

```diff
@@ -412,6 +412,15 @@ func DeleteUser(ctx *context.Context) {
 		return
 	}
 
+	// admin should not delete himself
+	if u.ID == ctx.Doer.ID {
+		ctx.Flash.Error(ctx.Tr("admin.users.cannot_delete_self"))
+		ctx.JSON(http.StatusOK, map[string]interface{}{
```

Comment:
`http.StatusOK`?
I'd say that it is not okay in the slightest, what about `http.BadRequest` instead?
Or do you have other status code suggestions?

Reply:
I think status code not usefull for ui request.

Reply:
Agreed, but it should still not be sent as a `good` request because it... isn't.

Reply:
> `http.StatusOK`? I'd say that it is not okay in the slightest, what about `http.BadRequest` instead? Or do you have other status code suggestions?

I just followed below lines.

Reply:
> Agreed, but it should still not be sent as a `good` request because it... isn't.

and the js handle logic need a 200 response to redirect. 

https://github.com/go-gitea/gitea/blob/ae6a52440ac47b8d0eed5875de9d801bc68c1a4a/web_src/js/features/common-global.js?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L222-L225